mono/metadata/locales.c

```diff
@@ -536,39 +542,50 @@ ves_icall_System_Globalization_CultureInfo_construct_internal_locale_from_specif
 	return ret;
 }
 */
+
+
 MonoBoolean
-ves_icall_System_Globalization_RegionInfo_construct_internal_region_from_lcid (MonoRegionInfo *this_obj,
+ves_icall_System_Globalization_RegionInfo_construct_internal_region_from_lcid (MonoRegionInfo *this_obj_raw,
 		gint lcid)
 {
+ 	ICALL_ENTRY();
+	gboolean res = FALSE;
+	LOCAL_HANDLE_PUSH_FRAME ();
+	HANDLE_DCL_TYPED (MonoRegionInfo, this_obj);
+
 	const RegionInfoEntry *ri;
 	
 	ri = region_info_entry_from_lcid (lcid);
 	if(ri == NULL)
-		return FALSE;
-
-	return construct_region (this_obj, ri);
-}
-
-MonoBoolean
-ves_icall_System_Globalization_RegionInfo_construct_internal_region_from_name (MonoRegionInfo *this_obj,
-		MonoString *name)
-{
-	const RegionInfoNameEntry *ne;
-	char *n;
-	
-	n = mono_string_to_utf8 (name);
-	ne = (const RegionInfoNameEntry *)mono_binary_search (n, region_name_entries, NUM_REGION_ENTRIES,
-		sizeof (RegionInfoNameEntry), region_name_locator);
-
-	if (ne == NULL) {
-		/*g_print ("ne (%s) is null\n", n);*/
-		g_free (n);
-		return FALSE;
-	}
-	g_free (n);
-
-	return construct_region (this_obj, &region_entries [ne->region_entry_index]);
-}
+		goto done;
+
+	res = construct_region (this_obj, ri);
+
+done:
+	LOCAL_HANDLE_POP_FRAME ();
+	ICALL_EXIT ();
+	return res;
+}
+
+MONO_ICALL_DEFINE(VAL(MonoBoolean, res),
+		  ves_icall_System_Globalization_RegionInfo_construct_internal_region_from_name,
```

Comment:
We can't change indentation because of the helper macro for declaring an icall.

If we do it, we'll erase the whole history of those functions and have some ugly looking code as the result.